nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -2485,7 +2485,8 @@ private Description matchDereference(
     if (baseExpressionSymbol != null) {
       if (baseExpressionSymbol.type.isPrimitive()
           || baseExpressionSymbol.getKind() == ElementKind.PACKAGE
-          || ElementUtils.isTypeElement(baseExpressionSymbol)) {
+          || ElementUtils.isTypeElement(baseExpressionSymbol)
+          || baseExpressionSymbol.getKind() == ElementKind.TYPE_PARAMETER) {
```

Comment:
Not quite.  The `isTypeElement` method only catches when the element is a [class, enum, interface, or annotation type](https://github.com/typetools/checker-framework/blob/a49161a016f2e9f0f2590d5f2c42fb8a1e9dfee4/javacutil/src/main/java/org/checkerframework/javacutil/ElementUtils.java#L817-L822); array types are not handled.  That case is handled by the line this PR adds to `mayBeNullExpr`.

For type variables, the expression itself is just an `IdentifierTree`.  We know it can't be null because it represents a type parameter; that is caught by the check on this line.